### DISCUSSION TITLE
Semver 7.3.* CVE-2022-2588

### DISCUSSION THREAD

**Discussion by dc185334** [typescript-eslint/typescript-eslint#7177] [Community Feedback]
@typescript-eslint/eslint-plugin
@typescript-eslint/typescript-estree
@typescript-eslint/utils

arise on CVE-2022-2588 for the docker security scanner (third-party software)

The reason is semver 7.3.7 that is manually overrided by my package.json next way:

```
  "overrides": {
    ...
    "@typescript-eslint/eslint-plugin": {
      "semver@7.3.7": "7.5.2"
    },
    "@typescript-eslint/typescript-estree": {
      "semver@7.3.7": "7.5.2"
    },
    "@typescript-eslint/utils": {
      "semver@7.3.7": "7.5.2"
    },
    ...
}```

Did use it for nestjs app and did not find any issue during dozen of internal and out own tests (unit/integ./e2e)

**Comment by bradzacher** [comment]
From the looks of [CVE-2022-2588](https://cve.report/CVE-2022-25883):
1) it's super duper low risk and low value
2) it relies on untrusted user data as input

We only pass data from either `eslint` `package.json` `version` field or from `typescript`'s exported `version` variable - both of which are highly trusted data.

Additionally our packages use a caret range which allows `7.5.2`:

https://github.com/typescript-eslint/typescript-eslint/blob/a8c824a1e84453f93cd2b464fc102bc878c1aff3/packages/eslint-plugin/package.json#L55
https://github.com/typescript-eslint/typescript-eslint/blob/a8c824a1e84453f93cd2b464fc102bc878c1aff3/packages/typescript-estree/package.json#L50
https://github.com/typescript-eslint/typescript-eslint/blob/a8c824a1e84453f93cd2b464fc102bc878c1aff3/packages/utils/package.json#L49

Which means that if you were to blow away your lock file - your package manager may choose to install a version like `7.5.2`.

Closing as there's no action for us.

**Comment by bradzacher** [comment]
As per our contributing guide - we do not use discussions for support requests or questions.
Our contributing guide contains information on better ways to ask questions: https://typescript-eslint.io/contributing/discussions
